### ISSUE TITLE
Not working with CommandQueue

### ISSUE THREAD

**Issue by brandonsturgeon** [programit/RoR2InstanceLootPlugin#1]
Hi there.

I prefer this mod to the alternatives. It's simple and it allows me more control over the spawnrates.

Currently, it doesn't appear to work with the popular [CommandQueue](https://thunderstore.io/package/KubeRoot/CommandQueue/) mod.

Any chance these two could be made to work together?

Thank you!

**Comment by programit** [comment]
Hey,

Are you mostly interested in the spawnrate portion of this plugin? If so I can definitely split that into it's own plugin that would work fine with CommandQueue. I haven't looked closely but I'd bet it's the portion of this plugin that provides per instance loot that is hooking the same methods as CommandQueue and causing issue. If you are interested in the instance loot as well though I can try to take a look and see if it's semi-simple fix

**Comment by brandonsturgeon** [comment]
Nope! The instancing is the main feature. The spawnrate adjustment is just a nice-to-have.

If you could find any easy way to make this mod compatible with CommandQueue, that would be **amazing**!

Right now we have to choose one or the other, but I know a number of people that would be thrilled to be able to use both.

I believe CommandQueue works with the other InstancedLoot mod: https://thunderstore.io/package/KubeRoot/InstancedLoot/
I prefer this one, though :)

Thank you!

**Comment by programit** [comment]
Ok so looked into it and the issue is that CommandQueue hooks a function I actually ensured never get called. The call path to get to that function normally spawns a network object which I want to avoid as a network object everyone can see and interact with which allows someone to potentially pick up someone else's items, instead I block that path and send a message to each client and have the client create a local command object which then only the local player sees and can interact with. Long story short the way I do that prevents CommandQueue's logic from ever running which is why it doesn't work.

That said the specific function they hook(PickupPickerController.OnDisplayBegin) I had duplicated in my code to display the picker dialog. Some quick testing locally seems to show it's safe to call their implementation instead as that function doesn't care if the object is a local or network object. Changing that things seem to work as expected.

I need to do some more testing. And there's another pr in progress to add some awesome balancing changes to the plugin. Hopefully we'll get that merged in the next day or two and then I'll create a pr for fixing this a day or two after once I can do a bit more testing.
